tests/system/test_introspection.py

```diff
@@ -136,3 +137,30 @@ async def test_controller_groups_and_parameters(sim_eiger_controller: EigerContr
                 assert any(param.key == key for param in parameters)
 
     await controller.connection.close()
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "sim_eiger_controller", [str(HERE / "eiger.yaml")], indirect=True
+)
+async def test_fetch_before_returning_parameters(
+    sim_eiger_controller: EigerController,
+):
+    controller = sim_eiger_controller
+    await controller.initialise()
+    detector_controller = controller.get_sub_controllers()["Detector"]
+    count_time_attr = detector_controller.attributes.get("count_time")
+    bit_depth_image_attr = detector_controller.attributes.get("bit_depth_image")
+    bit_depth_image_attr.updater.config_update = AsyncMock()
+    count_time_attr.updater.config_update = AsyncMock()
+    assert not detector_controller._parameter_updates
+    await count_time_attr.updater.put(detector_controller, count_time_attr, 2)
+    assert (
+        detector_controller._parameter_updates
+        and "bit_depth_image" not in detector_controller._parameter_updates
+        and "bit_depth_readout" not in detector_controller._parameter_updates
+    )
+    count_time_attr.updater.config_update.assert_not_called()
+    bit_depth_image_attr.updater.config_update.assert_called()
+    await detector_controller.update()
+    count_time_attr.updater.config_update.assert_called()
```

Comment:
Nit: It is a bit hard to read in one big block. I generally break things up into the initial setup and blocks of function calls with the relevant asserts.

```suggestion
    controller = sim_eiger_controller
    await controller.initialise()
    detector_controller = controller.get_sub_controllers()["Detector"]
    count_time_attr = detector_controller.attributes.get("count_time")
    bit_depth_image_attr = detector_controller.attributes.get("bit_depth_image")
    bit_depth_image_attr.updater.config_update = AsyncMock()
    count_time_attr.updater.config_update = AsyncMock()
    
    assert not detector_controller._parameter_updates
    await count_time_attr.updater.put(detector_controller, count_time_attr, 2)
    assert (
        detector_controller._parameter_updates
        and "bit_depth_image" not in detector_controller._parameter_updates
        and "bit_depth_readout" not in detector_controller._parameter_updates
    )
    count_time_attr.updater.config_update.assert_not_called()
    bit_depth_image_attr.updater.config_update.assert_called()
    
    await detector_controller.update()
    count_time_attr.updater.config_update.assert_called()
```